nemo_skills/code_execution/sandbox.py

```diff
@@ -253,6 +254,31 @@ async def is_proof_correct(self, pred_output, timeout=30.0):
             return "timeout"
         return determine_proof_status(output)
 
+    def _check_ready(self, timeout: float = 5.0) -> bool:
+        """Readiness check against the sandbox health endpoint."""
+        url = f"http://{self.host}:{self.port}/health"
+
+        if self.ssh_server and self.ssh_key_path:
+            import sshtunnel_requests
+
+            try:
+                sshtunnel_request = sshtunnel_requests.from_url(f"ssh://{self.ssh_server}:22", self.ssh_key_path)
```

Comment:
NIT: can you restructure so exception handling and status code parsing is only written once?